shopelectro/tests/tests_models.py

```diff
@@ -86,3 +86,32 @@ def test_get_brands_from_products(self):
                 p: p.tags.filter(group__name=settings.BRAND_TAG_GROUP_NAME).first()
                 for p in products
             }
+
+    def test_get_catalog_blocks(self):
+        """Perform four queries to fetch in batch blocks, categories, pages and category's children."""
+        roots = Category.objects.filter(level=0)
+        roots_count = roots.count()
+        for category in roots:
+            CatalogBlock.objects.create(category=category)
+
+        with self.assertNumQueries(4):
+            blocks = list(CatalogBlock.objects.blocks())
+            self.assertEquals(roots_count, len(blocks))
+            for block in blocks:
+                self.assertTrue(block.category)
+                self.assertTrue(block.category.page)
+                self.assertTrue(block.rows())
+
+
+@tag('fast')
+class CatalogBlockModel(TestCase):
+
+    fixtures = ['dump.json']
+
+    def test_rows_count(self):
+        first, second, *_  = Category.objects.all()
+        block = CatalogBlock.objects.create(category=first)
+        sized_block = CatalogBlock.objects.create(category=second, block_size=2)
+
+        self.assertNotEquals(first.children.active(), len(sized_block.rows()))
```

Comment:
you are comparing QS with number here. Let me suggest what you meaned

```suggestion
        self.assertEquals(first.children.active().count(), sized_block.rows().count())
```

Reply:
Sorry for the mess. I have reworked the test